packages/analytics/analytics-chart/src/composables/useExploreResultToDatasets.spec.ts

```diff
@@ -178,214 +178,300 @@ describe('useVitalsExploreDatasets', () => {
+    expect(result.value.datasets[4].backgroundColor).toEqual('#ffb6b6')
+  })
 
-it('Correct status code colors', () => {
-  const exploreResult: ComputedRef<ExploreResultV4> = computed(() => ({
-    data: [
-      {
-        timestamp: '2022-01-01T01:01:02Z',
-        event: {
-          status_code: 100,
-          metric1: 2,
-        },
-      } as GroupByResult,
-      {
-        timestamp: '2022-01-01T01:01:02Z',
-        event: {
-          status_code: 200,
-          metric1: 2,
-        },
-      } as GroupByResult,
-      {
-        timestamp: '2022-01-01T01:01:02Z',
-        event: {
-          status_code: 300,
-          metric1: 2,
-        },
-      } as GroupByResult,
-      {
-        timestamp: '2022-01-01T01:01:02Z',
-        event: {
-          status_code: 400,
-          metric1: 2,
-        },
-      } as GroupByResult,
-      {
-        timestamp: '2022-01-01T01:01:02Z',
-        event: {
-          status_code: 500,
-          metric1: 2,
-        },
-      } as GroupByResult,
-    ],
-    meta: {
-      start_ms: 1640998862000,
-      end_ms: 1640998870000,
-      granularity_ms: 86400000,
-      metric_names: ['metric1'] as any as ExploreAggregations[],
-      display: {
-        status_code: {
-          100: {
-            name: '100',
-            deleted: false,
+
+  it('sorts datasets accordingly', () => {
+    const exploreResult: ComputedRef<ExploreResultV4> = computed(() => ({
+      data: [
+        {
+          timestamp: '2022-01-01T01:01:02Z',
+          event: {
+            route: 'route1',
+            status_code: 100,
+            metric1: 1,
           },
-          200: {
-            name: '200',
-            deleted: false,
+        } as GroupByResult,
+        {
+          timestamp: '2022-01-01T01:01:02Z',
+          event: {
+            route: 'route1',
+            status_code: 200,
+            metric1: 2,
           },
-          300: {
-            name: '300',
-            deleted: false,
+        } as GroupByResult,
+        {
+          timestamp: '2022-01-01T01:01:02Z',
+          event: {
+            route: 'route2',
+            status_code: 100,
+            metric1: 3,
+          },
+        } as GroupByResult,
+        {
+          timestamp: '2022-01-01T01:01:02Z',
+          event: {
+            route: 'route2',
+            status_code: 200,
+            metric1: 4,
           },
-          400: {
-            name: '400',
-            deleted: false,
+        } as GroupByResult,
+        {
+          timestamp: '2022-01-01T01:01:02Z',
+          event: {
+            route: 'route3',
+            status_code: 100,
+            metric1: 5,
           },
-          500: {
-            name: '500',
-            deleted: false,
+        } as GroupByResult,
+        {
+          timestamp: '2022-01-01T01:01:02Z',
+          event: {
+            route: 'route3',
+            status_code: 200,
+            metric1: 6,
```

Comment:
I think that this doesn't actually test it, right?  In order to test it, we'd need 1 segment in route 1 or 2 to be larger than any 1 segment in route 3, but for the sum of segments in route 3 to be larger than the sum of segments in route 2.

Here, route 3 would be sorted first based on both the old criteria and the new criteria.